dev/archery/archery/integration/datagen.py

```diff
@@ -1401,9 +1401,8 @@ def generate_nested_case():
                            get_field('item', 'int32'), 4),
         StructField('struct_nullable', [get_field('f1', 'int32'),
                                         get_field('f2', 'utf8')]),
-        # Fails on Go (ARROW-8452)
-        # ListField('list_nonnullable', get_field('item', 'int32'),
-        #           nullable=False),
+        ListField('list_nonnullable', get_field('item', 'int32'),
+                  nullable=False),
```

Comment:
Perhaps make a separate `generate_non_nullable_list_case` so that other nested cases can still be checked on Go?

Reply:
Thanks a lot for your review, @pitrou !

all our integration test files are generated with a nullable and non-nullable version, which I assumed so far to make it a requirement to support nullable of a given type.

Maybe the "nested" is too broad, and we need to split it in different types.

I am parking this PR and will prioritize #12407 and #12409 as I am struggling to:
* run integration tests locally - it takes too much time.
* have a mechanism to add, remove and change tests

This is because the test file names are currently coupled to the names in the tests in the golden files. I am decoupling them in as part of #12409 to allow more flexibility in renaming, spliting, etc. test cases in this section.